File: EZcpp/chall.cpp
#include<bits/stdc++.h>
#include<windows.h>
using namespace std;

string en_flag = "mssctf{please_set_a_flag}";
string buf = "mssctf{This_is_a_fake_flag_hhhh}";

int main() {
    string input;
    
    cout << "[EZcpp] welcome to mssctf2021" << endl;
    cout << "please input your password and I will check it" <<endl;
    cout << "input : " ;
    cin >> input ;
    for (int i=0;i<32;i++){
        buf.at(i) = (input.at(i) ^ i + i) + 18;
    }
    if(buf == en_flag){
        cout << "Congratulations!!!" << endl;
        cout << "Here is your flag : mssctf{" << input << "}" << endl;
        Sleep(0xdeadbeef);
    }
    else{
        cout << "OHHH,you are wrong.Try again!" << endl;
        Sleep(0xdeadbeef);
    }
    return 0;
}

File: Strange_pyc/chall.py
import time

print("[Strange_pyc] Welcome to mssctf2021.")
magic = mssctf{please_set_a_flag}
hanser = input("please input your password : ")
ipt = list(hanser)
if len(ipt) != len(magic):
    print("QwQ , you are wrong . Try again ! ")
    time.sleep(114514)
    exit(114514)
for i in range(len(ipt)//2):
    ipt[i*2], ipt[i*2+1] = ipt[i*2+1], ipt[i*2]
for i in range(len(ipt)):
    ipt[i] = ord(ipt[i]) ^ 126
for i in range(len(ipt)):
    if ipt[i] != magic[i]:
        print("QwQ , you are wrong . Try again ! ")
        time.sleep(114514)
        exit(114514)
print("Congratulations!!!")
print("Here is your flag : " + "mssctf{" + hanser + "}")
time.sleep(114514)